e2e-tests/batch-mode-tests.sh

```diff
@@ -0,0 +1,107 @@
+echo "${print_prefix} Total openmrs test patients ---> ${total_test_patients}"
+curl -L -X GET -u admin:Admin123 --connect-timeout 5 --max-time 20 http://localhost:8099/openmrs/ws/fhir2/R4/Encounter/ 2>/dev/null >> ./omrs/encounters.json
+total_test_encounters=$(jq '.total' ./omrs/encounters.json)
+echo "${print_prefix} Total openmrs test encounters ---> ${total_test_encounters}"
+curl -L -X GET -u admin:Admin123 --connect-timeout 5 --max-time 20 http://localhost:8099/openmrs/ws/fhir2/R4/Observation/ 2>/dev/null >> ./omrs/obs.json
+total_test_obs=$(jq '.total' ./omrs/obs.json)
+echo "${print_prefix} Total openmrs test obs ---> ${total_test_obs}"
+echo "${print_prefix} Counting number of patients, encounters and obs sinked to parquet files"
+total_patients_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount ./Patient/ | awk '{print $3}')
+echo "${print_prefix} Total patients synched to parquet ---> ${total_patients_streamed}"
+total_encounters_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount ./Encounter/ | awk '{print $3}')
+echo "${print_prefix} Total encounters synched to parquet ---> ${total_encounters_streamed}"
+total_obs_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount ./Observation/ | awk '{print $3}')
+echo "${print_prefix} Total obs synched to parquet ---> ${total_obs_streamed}"
+if [[ ${total_patients_streamed} == ${total_test_patients} && ${total_encounters_streamed} == ${total_test_encounters} && $total_obs_streamed == ${total_test_obs} ]] ;
+then
+  echo "${print_prefix} BATCH MODE WITH PARQUET SYNC EXECUTED SUCCESSFULLY"
+else
+  echo "${print_prefix} BATCH MODE WITH PARQUET SYNC TEST FAILED"
+  exit 1
+fi
+cd "${current_path}"
+echo "${print_prefix} RUNNING BATCH MODE WITH FHIR SYNC"
+echo "${print_prefix} STARTING FHIR SINK SERVER"
+docker-compose -f sink-compose.yml up -d
+fhir_server_start_wait_time=0
+fhir_server_status_code=$(curl -o /dev/null --head -w "%{http_code}" -L -X GET -u hapi:hapi --connect-timeout 5 --max-time 20 http://localhost:8098/fhir/Observation 2>/dev/null)
+until [[ ${fhir_server_status_code} -eq 200 ]]; do
+      sleep 1s;
+      echo "${print_prefix} WAITING FOR FHIR SERVER TO START"
+      fhir_server_status_code=$(curl -o /dev/null --head -w "%{http_code}" -L -X GET -u hapi:hapi --connect-timeout 5 --max-time 20 http://localhost:8098/fhir/Observation 2>/dev/null)
+      ((fhir_server_start_wait_time+=1))
+      if [[ fhir_server_start_wait_time == 10 ]]; then
+        echo "${print_prefix} TERMINATING AS FHIR SERVER TOOK TOO LONG TO START"
+        exit 1
+      fi
+done
+echo "${print_prefix} FHIR SERVER STARTED SUCCESSFULLY"
+echo "mvn exec:java -pl batch
+ -Dexec.args=--openmrsServerUrl=http://localhost:8099/openmrs --openmrsUserName=admin --openmrsPassword=Admin123
+   --searchList=Patient --batchSize=20 --targetParallelism=20
+   --fhirSinkPath=http://localhost:8098/fhir
+   --sinkUserName=hapi --sinkPassword=hapi"
+
+mvn exec:java -pl batch \
```

Comment:
nit: Any particular reason that here you are running the batch pipeline through `mvn exec:java` but on line 27 with `java -cp`. I don't mind either but I prefer to be consistent.